core/cmd/remote_client.go

```diff
@@ -68,16 +68,20 @@ func (cli *Client) CreateServiceAgreement(c *clipkg.Context) error {
 
 // CreateExternalInitiator adds an external initiator
 func (cli *Client) CreateExternalInitiator(c *clipkg.Context) error {
-	if c.NArg() != 2 {
-		return cli.errorOut(errors.New("create expects two arguments: a name and a url"))
+	if c.NArg() != 1 && c.NArg() != 2 {
+		return cli.errorOut(errors.New("create expects 1 - 2 arguments: a name and a url (optional)"))
 	}
 
 	var request models.ExternalInitiatorRequest
 	request.Name = c.Args().Get(0)
-	if url, err := url.ParseRequestURI(c.Args().Get(1)); err == nil {
-		request.URL = models.WebURL(*url)
-	} else {
-		return cli.errorOut(err)
+
+	// process optional URL
+	if c.NArg() == 2 {
+		if url, err := url.ParseRequestURI(c.Args().Get(1)); err == nil {
```

Comment:
Idiomatic Golang nitpick: [indent the error flow](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow) and let keep the happy path unindented.